text/0000-std-rpc-api.md

```diff
@@ -0,0 +1,397 @@
+- Feature Name: `std-rpc-api`
+- Start Date: 2020-11-02
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Existing MLAPI [**Convenience** and **Performance** RPCs](https://mlapi.network/wiki/messaging-system/) are sub-optimal in terms of required boilerplate code, API design and runtime performance.
+
+This RFC proposes a new **Standard** RPC API that excels at minimal boilerplate, API design/clarity, extensibility (future-proofing) and runtime performance.
+
+The initial implementation of this standard would be kept minimal and further-extended with additional features down the line as it will also support future extensions and enhancements without changing the standard API necessarily.
+
+# Motivation
+[motivation]: #motivation
+
+Achieve better performance, unify & standardize RPC API, make RPC API extensible & future-proof and reduce boilerplate code.
+
+Also a simple internal performance benchmark came with these results:
+
+    Local RPC Send Performance Test (524280 calls)
+    Convenience -> 4361.0433ms
+    Performance -> 2607.7266ms
+    Standard -> 2205.043ms
+    Local RPC Recv Performance Test (524280 calls)
+    Convenience -> 9626.3536ms
+    Performance -> 5923.7964ms
+    Standard -> 1935.296ms
+
+We're looking at **up to 5.24x better performance** (compared to convenience API vs standard API receive performance).
+
+Beyond that, on an internal project, we saw roughly **%80 less boilerplate code** compared to performance RPCs and standard RPCs which is also a huge win.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Multiplayer framework provides 2 main network constructs ([ServerRpc](#serverrpc) and [ClientRpc](#clientrpc)) to execute logic on either server-side or client-side. This concept often called as [Remote Procedure Call (RPC)](https://en.wikipedia.org/wiki/Remote_procedure_call) and has wide adoption across the industry.
+
+## RPC Methods
```

Comment:
I cant find the subtitle about response.

If this is implemented, i'd really like to see a way to wait for the result. Unity itself uses Coroutines a lot, but my personal preference is an async task based API.

Reply:
@TJHeuvel, I don't think it really makes sense to use async tasks for this, since messages are being polled during frames. Using async for this would be kind of pointless since it wouldn't receive the message any quicker. If you really need to, spin off a task during the coroutine.